datashader/bundling.py

```diff
@@ -442,7 +444,10 @@ class hammer_bundle(directly_connect_edges):
         Column name for each edge weight. If None, weights are ignored.""")
 
     def __call__(self, nodes, edges, **params):
+        # Merge user-provided parameters with local parameters
         p = param.ParamOverrides(self, params)
+        for k in p:
+            setattr(self, k, p[k])
```

Comment:
Oops; I didn't see these diffs earlier, only the later ones.  This isn't the right approach, as it will be very confusing for users to see arguments they supply to a call being the new persistent values. Definitely would be a big source of confusing bugs for users.  Instead, would need to pass the ``p`` object to the method below that uses it.  Alas, I think that renders my previous suggestion moot and counterproductive -- there is no increase in simplicity by making these functions into methods if they still have to access the parameters via a supplied object.  So I guess this whole commit should be reverted?  Sorry about that; entirely my fault for not looking at the context in which those methods were being used.